Cmr 183

dseelig opened the pull request.

dseelig committed:
Install ReportingDB with ansible on a virtual machine.

Update postgresql library to version 42.1.4 in pom.xml.
Logging database configs in the PostgrePersistenceService.java.
Change the default.properties file load in the ReportingPrperties concstructor.
The default.properties is saved out of the deploy path.

dseelig committed:
Install JDK8 with openjdk8.
Use jdk8 and tomcat8 only.
Delete the test war file. Use war file from your local project.

dseelig committed:
Change vagrant and ansible playbook file from debian 8 jessie to debian 9.3.0 stretch.

claussni commented on vagrant/Vagrantfile:
```diff
@@ -0,0 +1,26 @@
+$script = <<SCRIPT
```
This $script variable has no use.

claussni commented on vagrant/ansible/hosts:
```diff
@@ -0,0 +1,2 @@
+[conductr]
+192.168.42.21
```
What is this IP?

claussni commented on vagrant/ansible/playbook.yml:
```diff
@@ -0,0 +1,156 @@
+      dest: "/home"
+
+  - name: Create Tables
+    bocome: true
```
This typo leads to ansible error: `bocome` should be `become`

claussni commented:
Vagrant run works. However, Tomcat log shows some errors. One is:

19-Dec-2017 12:04:21.242 WARNING [localhost-startStop-1] org.apache.tomcat.util.scan.StandardJarScanner.scan Failed to scan [file:/usr/share/java/jsp-api-2.3.jar] from classloader hierarchy
 java.io.FileNotFoundException: /usr/share/java/jsp-api-2.3.jar (No such file or directory)
        at java.util.zip.ZipFile.open(Native Method)
        at java.util.zip.ZipFile.<init>(ZipFile.java:225)
        at java.util.zip.ZipFile.<init>(ZipFile.java:155)
        at java.util.jar.JarFile.<init>(JarFile.java:166)
        at java.util.jar.JarFile.<init>(JarFile.java:130)
        at org.apache.tomcat.util.scan.JarFileUrlJar.<init>(JarFileUrlJar.java:60)
        at org.apache.tomcat.util.scan.JarFactory.newInstance(JarFactory.java:49)
        at org.apache.tomcat.util.scan.StandardJarScanner.process(StandardJarScanner.java:338)
        at org.apache.tomcat.util.scan.StandardJarScanner.scan(StandardJarScanner.java:288)
        at org.apache.jasper.servlet.TldScanner.scanJars(TldScanner.java:262)
        at org.apache.jasper.servlet.TldScanner.scan(TldScanner.java:104)
        at org.apache.jasper.servlet.JasperInitializer.onStartup(JasperInitializer.java:101)
        at org.apache.catalina.core.StandardContext.startInternal(StandardContext.java:5196)
        at org.apache.catalina.util.LifecycleBase.start(LifecycleBase.java:150)
        at org.apache.catalina.core.ContainerBase.addChildInternal(ContainerBase.java:752)
        at org.apache.catalina.core.ContainerBase.addChild(ContainerBase.java:728)
        at org.apache.catalina.core.StandardHost.addChild(StandardHost.java:734)
        at org.apache.catalina.startup.HostConfig.deployDirectory(HostConfig.java:1107)
        at org.apache.catalina.startup.HostConfig$DeployDirectory.run(HostConfig.java:1841)
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)

Avoidable?

claussni commented on .gitignore:
```diff
@@ -1,2 +1,11 @@
 /target/
 /src/main/resources/local.properties
+vagrant/ansible/files/qucosa-fcrepo-reportingdb-1.0.0-SNAPSHOT.war
```
The war-file ignore needs a wildcard. Otherwise versions different from 1.0.0-SNAPSHOT are going to be reported as change.

claussni commented on .gitignore:
```diff
@@ -1,2 +1,11 @@
+vagrant/ansible/files/qucosa-fcrepo-reportingdb-1.0.0-SNAPSHOT.war
+.classpath
+.project
+.settings/
```
.classpath, .project, build.properties and .settings/ are IDE specific folder and should not be excluded here.
Better to configure these in your global .gitignore configuration:
https://help.github.com/articles/ignoring-files/#create-a-global-gitignore

claussni commented on .gitignore:
```diff
@@ -1,2 +1,11 @@
+.settings/
+META-INF/
+build.properties
+doc/HarvesterResumptionLogic.ods
```
Why is this in ignore?

claussni commented on vagrant/ansible/playbook.yml:
```diff
@@ -0,0 +1,156 @@
+  tasks:
+  
+  - name: Install python2.7
+    raw: "sudo apt-get update -qq && sudo apt-get install -qq python2.7 aptitude"
```
Why using a raw command instead of `package`?

The `package` declaration is independent of a particular package manager: http://docs.ansible.com/ansible/latest/package_module.html#package-generic-os-package-manager

claussni commented on vagrant/ansible/playbook.yml:
```diff
@@ -0,0 +1,156 @@
+    package: name=unzip state=latest
+    
+  - name: Install python3-software-properties
+    command: apt-get install -y python3-software-properties
```
Shouldn't `package python3-software-proerties` work as well?

claussni commented on vagrant/ansible/playbook.yml:
```diff
@@ -0,0 +1,156 @@
+      src: "templates/tomcat-users.xml"
+      dest: "/etc/tomcat8/tomcat-users.xml"
+  - name: create reporting folder
+    command: mkdir /var/lib/tomcat8/webapps/qucosa-fcrepo-reportingdb
```
This works, however, there are nicer approaches for installing a web application to tomcat.

This way is not the way it's done on the production machines.

claussni commented on vagrant/ansible/playbook.yml:
```diff
@@ -0,0 +1,156 @@
+      mode: 0755
+
+  - name: unzip deployed war
+    unarchive: src=/var/lib/tomcat8/webapps/qucosa-fcrepo-reportingdb-1.0.0-SNAPSHOT.war dest=/var/lib/tomcat8/webapps/qucosa-fcrepo-reportingdb copy=no mode=0755
```
Unpacking WAR files is not necessary. Tomcat can handle this.

claussni commented on vagrant/ansible/files/config/default.properties:
```diff
@@ -0,0 +1,28 @@
+db.driver=org.postgresql.Driver
+db.user=reportingDB
+db.passwd=reportingDB
+oai.url=http://sdvcmr-prod-core01.slub-dresden.de:8080/fedora/oai
```
Although the system host will not work from outside, those hostnames should never appear in public repositories.
Consider using default hosts like `localhost` and replace it via external configuration at ansible run.

See http://docs.ansible.com/ansible/latest/playbooks_variables.html#variable-file-separation:

> It’s a great idea to keep your playbooks under source control, but you may wish to make the playbook source public while keeping certain important variables private. Similarly, sometimes you may just want to keep certain information in different files, away from the main playbook.

dseelig committed:
Change the ansible playbook configuration.
The file new-playbook.yml is an puffer file and deleted if the ansible config finished.

dseelig committed:
Change the ansible playbook with new routines.
Add system vars json config.
Modified java classes for tests.

claussni committed:
Fix missing WAR file name in tomcat context configuration

claussni committed:
Please continue testing with test system

claussni committed:
Remove deprecated sudo option

claussni committed:
Install packages via `package` module

claussni committed:
Make Tomcat service directory owner

Otherwise the service will fail to start due to insufficient write privileges.

claussni committed:
Update .gitignore

claussni committed:
Do not install `unzip`

claussni committed:
Remove absolute paths

claussni committed:
Become root before adding an apt_key

claussni committed:
Fix ansible setup

- Install postgreSQL 9.6 from Debian 9 repositories
- Restart Tomcat via handler after deploying application
- Use `package` for installation wherever possible

claussni committed:
Run Ansible provisioner in verbose mode

claussni committed:
Write config files before creating database

claussni committed:
Ensure PostgreSQL service is running

claussni committed:
Ensure user does not have unnecessary privileges

claussni committed:
Make ansible less verbose

claussni committed:
Grant access to database from other hosts

dseelig committed:
Modified pom.xml with resources filtering.
Change ansible playbook with project variables from mvn project build.

dseelig committed:
Remove old deployment handling from playbook.

dseelig committed:
Remove ansible folders and files from project.
Change the vagrant file and use the extern saved ansible structure.
Change the pom.xml and add the resources plugin for build config files which use by ansible.

dseelig committed:
Remove the readme.txt because extract the ansible structure from project.
Remove the ansible config in the vagrant file.

dseelig committed:
Change file path to property file.
